src/annotator/config/index.js

```diff
@@ -1,53 +1,240 @@
+        valueFn: this.settings.hostPageSetting,
+      },
+      group: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: () => this.settings.group,
+      },
+      focus: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      theme: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      usernameUrl: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      onLayoutChange: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      openSidebar: {
+        allowInBrowserExt: true,
+        defaultValue: false,
+        coerce: toBoolean,
+        valueFn: this.settings.hostPageSetting,
+      },
+      query: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: () => this.settings.query,
+      },
+      requestConfigFromFrame: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      services: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      showHighlights: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: () => this.settings.showHighlights,
+      },
+      notebookAppUrl: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: () => this.settings.notebookAppUrl,
+      },
+      sidebarAppUrl: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: () => this.settings.sidebarAppUrl,
+      },
+      // Sub-frame identifier given when a frame is being embedded into
+      // by a top level client
+      subFrameIdentifier: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      externalContainerSelector: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+    };
+  }
+
+  annotatorDefinitions() {
+    return this.configDefinitions();
+  }
+
+  sidebarDefinitions() {
+    return this.configDefinitions();
+  }
+
+  notebookDefinitions() {
+    const config = this.configDefinitions();
+    // Don't pass `annotations` setting to notebook.
+    delete config.annotations;
+    return config;
+  }
+
+  /**
+   * Process the config definition to create the final config value.
+   *   - find the value from its source (valueFn)
+   *   - filter allowInBrowserExt as needed
+   *   - coerce the value if necessary
+   *   - set a default if the value is undefined
+   * @param {string} name - Config key name
+   * @param {ConfigDefinition} configDef
+   */
+  createValueFromDefinition(name, configDef) {
```

Comment:
This method (and perhaps some others) appear like they might be private methods.